validator/sawtooth_validator/journal/completer.py

```diff
@@ -27,38 +27,51 @@
 from sawtooth_validator.networking.dispatch import Handler
 from sawtooth_validator.networking.dispatch import HandlerResult
 from sawtooth_validator.networking.dispatch import HandlerStatus
+from sawtooth_validator.journal.block_wrapper import NULL_BLOCK_IDENTIFIER
 
 LOGGER = logging.getLogger(__name__)
 
 
 class Completer(object):
-    def __init__(self, block_store):
+    def __init__(self, block_store, gossip):
+        self.gossip = gossip
         self.batch_cache = TimedCache()
         self.block_cache = BlockCache(block_store)
+        # avoid throwing away the genesis block
+        self.block_cache[NULL_BLOCK_IDENTIFIER] = None
         self._on_block_received = None
         self._on_batch_received = None
 
     def _check_block(self, block):
         # currently only accepting finalized blocks
         # in the future if the blocks will be built
```

Comment:
I know this isn't part of your change, but this comment is a sentence fragment. Recommend changing this to docstring which describes what _check_block does.